packages/plugin-search/src/Search/index.ts

```diff
@@ -24,46 +25,45 @@ export const generateSearchCollection = (searchConfig: SearchConfig): Collection
       fields: [
         {
           name: 'title',
+          type: 'text',
           admin: {
             readOnly: true,
           },
-          type: 'text',
         },
         {
           name: 'priority',
+          type: 'number',
           admin: {
             position: 'sidebar',
           },
-          type: 'number',
         },
         {
           name: 'doc',
+          type: 'relationship',
           admin: {
             position: 'sidebar',
             readOnly: true,
           },
           index: true,
-          maxDepth: 0,
```

Comment:
Feels like we just allow overrides on this entire field, instead of piecemealing each property one by one.

Reply:
Let's do it, allow overrides on the whole field. Probably only need the `doc` field to be able to be overridden for now.

```ts
    fieldOverrides: {
      doc:  {
        maxDepth: 0,
      },
    },
```